example-get-started/code/.github/workflows/cml.yaml

```diff
@@ -8,6 +8,7 @@ jobs:
       - uses: iterative/setup-dvc@v1
       - uses: actions/checkout@v3
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
@tasdomas Q: why do we need this?

Reply:
For `cml comment create --pr` (comments on PRs) to work, cml needs to be able to determine the PR id based on the HEAD commit. However github runs workflows on a {master..PR} merge commit, which cml currently cannot attribute to the actual PR. This is a temporary workaround until we implement proper PR lookup in cml.

Reply:
got it. In this repo we were also publishing comments to the main branch commits:

https://github.com/iterative/example-repos-dev/pull/140/files#diff-74446cb71a98e9596cef50d831916c75ed4f0c65a5083c75dca8de15cdd5dd50R18-R22

what will happen to those with this new change (and with `--pr`)?

I mean reports like this https://github.com/iterative/example-get-started/commit/a470fae5c0ccf71641e10d99e69e97534c3d811e#commitcomment-84532489

Reply:
There's an issue with commit PRs, that affects CML - in github they [no longer appear in the PR timeline](https://github.blog/changelog/2022-08-04-commit-comments-no-longer-appear-in-the-pull-request-timeline/) which may make adding commit comments in our example repos fall short in showcasing CML functionality.

@casperdcl  - what do you think?

Reply:
> In this repo we were also publishing comments to the main branch commits

afaik `--pr` should fallback to non-`--pr` in such cases.

Alternatively should we change this workflow from `on: push` to `on: pull_request` (or both `push` & `pull_request`)? To me the `main` monitoring use case is different from the PR use case, so maybe even separate workflows?

Reply:
yep, would be a bit annoying to duplicate the workflow file in this repo in this simple case though

Reply:
I've revisited this and if all we want is commit tracking, then we don't need this change. Sorry, I was a tad over-zealous.